.github/sync-repo-settings.yaml

```
@@ -0,0 +1,40 @@
+# Rules for main branch protection
+branchProtectionRules:
+# Identifies the protection rule pattern. Name of the branch to be protected.
+# Defaults to `main`
+- pattern: main
+  # Can admins overwrite branch protection.
+  # Defaults to `true`
+  isAdminEnforced: true
+  # Number of approving reviews required to update matching branches.
+  # Defaults to `1`
+  requiredApprovingReviewCount: 1
+  # Are reviews from code owners required to update matching branches.
+  # Defaults to `false`
+  requiresCodeOwnerReviews: true
+  # Require up to date branches
+  requiresStrictStatusChecks: false
+  # List of required status check contexts that must pass for commits to be accepted to matching branches.
+  requiredStatusCheckContexts:
+    - 'Kokoro'
+    - 'Kokoro system-3.8'
+    - 'cla/google'
+    - 'OwlBot Post Processor'
+    - 'unittest / cover (pull_request)'
+    - 'docs / docfx (pull_request)'
+    - 'docs / docs (pull_request)'
+    - 'lint / lint (pull_request)'
```

```suggestion
    - 'docs'
    - 'docfx'
    - 'lint'
    - 'unit (3.7)'
    - 'unit (3.8)'
    - 'unit (3.9)'
    - 'unit (3.10)'
    - 'unit (3.11)'
    - 'cover'
    - 'mypy'
```

See similar format used in https://github.com/googleapis/python-bigquery-pandas/blob/main/.github/sync-repo-settings.yaml. I added `mypy` as well. Please consider adding samples presubmits as required also to ensure they are kept up to date.